src/ucp/core/ucp_worker.c

```diff
@@ -2731,3 +2738,28 @@ void ucp_worker_discard_uct_ep(ucp_worker_h worker, uct_ep_h uct_ep,
 
     ucp_worker_discard_tl_uct_ep(worker, uct_ep, ep_flush_flags);
 }
+
+static int ucp_worker_discard_remove_filter(const ucs_callbackq_elem_t *elem,
+                                            void *arg)
+{
+    return (elem->cb == ucp_worker_discard_uct_ep_progress) ||
+           (elem->cb == ucp_worker_discard_uct_ep_destroy_progress);
+}
+
+static void ucp_worker_discard_cleanup(ucp_worker_h worker)
+{
+    uct_ep_h uct_ep;
+    ucp_request_t *req;
+
+    UCS_ASYNC_BLOCK(&worker->async);
+    ucs_callbackq_remove_if(&worker->uct->progress_q,
+                            ucp_worker_discard_remove_filter, NULL);
+    kh_foreach(&worker->discard_uct_ep_hash, uct_ep, req, {
+        --worker->flush_ops_count;
```

Comment:
is it always correctly to decrement the counter? can we assert it somehow to be sure?

Reply:
yes, if something in the hash, it is ok to decrement

Reply:
added wrappers that checks that `flush_ops_count = { 0 .. UINT_MAX }`